File: ironman/src/main/java/com/avengers/ironman/largeimage/LargeImageInfo.java
package com.avengers.ironman.largeimage;

public class LargeImageInfo {
    private String framework = "other";
    private String url;
    private double fileSize;
    private double memorySize;
    private int width;
    private int height;
    private String activity;
    private String layoutLevel;
    private int layoutId;
    private String id;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getFramework() {
        return framework;
    }

    public void setFramework(String framework) {
        this.framework = framework;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public double getFileSize() {
        return fileSize;
    }

    public void setFileSize(double fileSize) {
        this.fileSize = fileSize;
    }

    public double getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(double memorySize) {
        this.memorySize = memorySize;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public String getActivity() {
        return activity;
    }

    public void setActivity(String activity) {
        this.activity = activity;
    }

    public String getLayoutLevel() {
        return layoutLevel;
    }

    public void setLayoutLevel(String layoutLevel) {
        this.layoutLevel = layoutLevel;
    }

    public int getLayoutId() {
        return layoutId;
    }

    public void setLayoutId(int layoutId) {
        this.layoutId = layoutId;
    }

    @Override
    public String toString() {
        return "LargeImageInfo{" +
                "framework='" + framework + '\'' +
                ", url='" + url + '\'' +
                ", fileSize=" + fileSize +
                ", memorySize=" + memorySize +
                ", width=" + width +
                ", height=" + height +
                ", activity='" + activity + '\'' +
                ", layoutLevel='" + layoutLevel + '\'' +
                ", layoutId='" + layoutId + '\'' +
                '}';
    }
}

File: ironman/src/main/java/com/avengers/ironman/largeimage/LargeImageInfo.kt
package com.avengers.ironman.largeimage

import android.graphics.Bitmap

class LargeImageInfo {
    var id: String? = ""
    var framework = "other"  //框架
    var url: String? = null
    var fileSize = 0.0 //文件大小
    var memorySize = 0.0 //内存大小
    var width = 0  //图片宽高
    var height = 0
    var viewWidth = 0 //控件宽高
    var viewHeight = 0
    var activity: String? = null //控件所在的activity名字
    var layoutLevel: String? = null //布局层级
    var viewString = "" //控件信息
    var bitmap: Bitmap? = null
    var phoneModel: String? = null //手机型号
    var phoneSysVersion: String? = null  //手机型号
    var sdcardSpace: String? = null  //sd卡剩余空间
    var romSpace: String? = null  //系统剩余空间
    var memorySpace: String? = null //内存占用

    override fun toString(): String {
        return "LargeImageInfo{" +
                "framework='" + framework + '\'' +
                ", url='" + url + '\'' +
                ", fileSize=" + fileSize +
                ", memorySize=" + memorySize +
                ", width=" + width +
                ", height=" + height +
                ", viewWidth=" + viewWidth +
                ", viewHeight=" + viewHeight +
                ", activity='" + activity + '\'' +
                ", layoutLevel='" + layoutLevel + '\'' +
                ", viewString=" + viewString +
                '}'
    }
}

fun largeImageInfoDsl(init: LargeImageInfo.() -> Unit): LargeImageInfo {
    val info = LargeImageInfo()
    info.init()
    return info
}